prow/config/config_test.go

```diff
@@ -3584,12 +3653,369 @@ func TestRefGetterForGitHubPullRequest(t *testing.T) {
+  config:
+    grace_period: 30s
+- repo: "^org/foo$"
+  cluster: "^trusted$"
+  config:
+    grace_period: 1m
+`),
+			expected: []*DefaultDecorationConfigEntry{
+				{
+					Repo:       regexp.MustCompile(""),
+					RepoRaw:    "",
+					Cluster:    regexp.MustCompile(""),
+					ClusterRaw: "",
+					Config: &prowapi.DecorationConfig{
+						Timeout:     &prowapi.Duration{Duration: 2 * time.Hour},
+						GracePeriod: &prowapi.Duration{Duration: 15 * time.Second},
+						UtilityImages: &prowapi.UtilityImages{
+							CloneRefs:  "clonerefs:default",
+							InitUpload: "initupload:default",
+							Entrypoint: "entrypoint:default",
+							Sidecar:    "sidecar:default",
+						},
+						GCSConfiguration: &prowapi.GCSConfiguration{
+							Bucket:       "default-bucket",
+							PathStrategy: prowapi.PathStrategyLegacy,
+							DefaultOrg:   "kubernetes",
+							DefaultRepo:  "kubernetes",
+						},
+						GCSCredentialsSecret: pStr("default-service-account"),
+					},
+				},
+				{
+					Repo:       regexp.MustCompile("^org/"),
+					RepoRaw:    "^org/",
+					Cluster:    regexp.MustCompile(""),
+					ClusterRaw: "",
+					Config: &prowapi.DecorationConfig{
+						Timeout: &prowapi.Duration{Duration: 1 * time.Hour},
+					},
+				},
+				{
+					Repo:       regexp.MustCompile("^org/repo$"),
+					RepoRaw:    "^org/repo$",
+					Cluster:    regexp.MustCompile(""),
+					ClusterRaw: "",
+					Config: &prowapi.DecorationConfig{
+						Timeout: &prowapi.Duration{Duration: 3 * time.Hour},
+					},
+				},
+				{
+					Repo:       regexp.MustCompile(""),
+					RepoRaw:    "",
+					Cluster:    regexp.MustCompile("^trusted$"),
+					ClusterRaw: "^trusted$",
+					Config: &prowapi.DecorationConfig{
+						GracePeriod: &prowapi.Duration{Duration: 30 * time.Second},
+					},
+				},
+				{
+					Repo:       regexp.MustCompile("^org/foo$"),
+					RepoRaw:    "^org/foo$",
+					Cluster:    regexp.MustCompile("^trusted$"),
+					ClusterRaw: "^trusted$",
+					Config: &prowapi.DecorationConfig{
+						GracePeriod: &prowapi.Duration{Duration: 1 * time.Minute},
+					},
+				},
+			},
+		},
+	}
+
+	for i := range tcs {
+		tc := tcs[i]
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			raw := json.RawMessage(tc.raw)
+			p := Plank{
+				DefaultDecorationConfigsRaw: &raw,
+			}
+			if err := p.FinalizeDefaultDecorationConfigs(); err != nil {
+				t.Errorf("error finalizing DefaultDecorationConfigs: %w", err)
+			}
+			if diff := cmp.Diff(tc.expected, p.DefaultDecorationConfigs, cmpopts.IgnoreUnexported(regexp.Regexp{})); diff != "" {
+				t.Errorf("expected result diff: %s", diff)
+			}
+		})
+	}
+}
+
+// complexConfig is shared by multiple test cases that test DefaultDecorationConfig
+// merging logic. It configures the upload bucket based on the org/repo and
+// uses either a GCS secret or k8s SA depending on the cluster.
+// A specific 'override' org overrides some fields in the trusted cluster only.
+var complexConfig = &Config{
```

Comment:
small nit: Its nicer to put something like this behind a getter func (`func complexConfig() *Config`) to avoid sharing state between tests